woocommerce/woocommerce-gutenberg-products-block
# Make Attribute Filter block compatible with PHP rendered Classic Template block

Update the Filter Attribute block to work with the Classic Template block which is PHP rendered, but also still maintain compatibility with the All Products block.

Filter Products by Attribute block will only work with the PHP rendered Classic Template block if the filters are applied to the URL and the page is re-rendered. This behaviour happens with the Classic Themes such as Storefront.




Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6136

### Testing

#### Automated Tests
 
### Manual Testing

How to test the changes in this Pull Request:

1. With a block theme, go to Appearance > Editor > Template > Product Catalog.
2. Add the Filter Products by Attribute block above the product grid.
3. Check some of the attributes on the frontend
4. Notice the page reloads and the list of products updates accordingly.
5. Now add the "Active Filters" block and repeat steps 1-4 and view the active filters appear upon page reload. Please bare in mind the "Clear All" functionality will not work with the Active Filters block yet, this needs to be worked on separately.
6. Add a test page to your site with the `All Products` block, and the `Filter by Attribute` block and ensure no regressions have been introduced here.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1.
2.
3.

### Changelog

> Allow adding the Filter Products by Attribute block to Product Catalog templates to filter products.
